# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../icc/pkg_fileutils_sysmacros.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./src/mknod.c.orig	Fri Jun 21 18:39:46 2002
+++ ./src/mknod.c	Fri Jun 21 18:40:16 2002
@@ -27,6 +27,14 @@
 
    David MacKenzie <djm@ai.mit.edu>  */
 
+#ifdef __INTEL_COMPILER
+#  define _SYS_SYSMACROS_H 1
+#  define major(dev) ((int)(((dev) >> 8) & 0xff))
+#  define minor(dev) ((int)((dev) & 0xff))
+#  define makedev(major, minor) ((((unsigned int) (major)) << 8) \
+                                | ((unsigned int) (minor)))
+#endif
+
 #include <config.h>
 #include <stdio.h>
 #include <getopt.h>
